controller/login_test.go

```diff
@@ -186,3 +199,233 @@ func (t TestLoginService) LinkSession(ctx *app.LinksessionLoginContext, brokerEn
 func (t TestLoginService) LinkCallback(ctx *app.LinkcallbackLoginContext, brokerEndpoint string, clientID string) error {
 	return ctx.TemporaryRedirect()
 }
+
+/*  For recent spaces test */
+
+type TestRecentSpacesREST struct {
```

Comment:
I think it's better to move this test to a separate file. login_recent_spaces_test.go or something.